### ISSUE TITLE
feature request: a way to pass a new color palette to an already created addPolygons() layer

### ISSUE THREAD

**Issue by psarnow** [rstudio/leaflet#170]
What I would like to do is change the vector of colors based on a user's input. For example, they might select a different variable to color each polygon by generating a new gradient.

Shiny allows us to do this, as if I change the color vector based on an input widget, the reactive function refreshes, and recreates the map based on the new vector. The problem is that with large shapefile objects, this refresh takes a long time.

Is there any way to change the color of the currently rendered shapes directly, without recreating the entire layer? It seems like the color argument is locked within the leaflet() function. Is there any other way to get at it?

**Comment by cholcomb** [comment]
This is what I was aiming for with rstudio/leaflet#163 (for GeoJSON). The idea here is to style based on data that's not in the original GeoJSON/Polygon object (i.e., styles based on dynamic data, which I believe is really the use case for R as opposed to just setting up a non-interactive Website). As discussed on the line note viewable @ https://github.com/rstudio/leaflet/pull/163#discussion_r37116951, I believe there might be a way to have something like styleGeoJSON/stylePolygons that just restyles based on original data, but I'm not yet certain if this is possible or high priority. 

In other words, you may want to test out this PR and see if it does what you're thinking. If you adopt it to your data, you could drop all of the properties except the id's in order to shrink the initial file size (and keep them in a data.frame used for changing the colors as in the example shiny_interactions.R). You will see that it's not that fast yet because it handles each feature separately but at least the lat-lngs aren't passed to the client with every change.

**Comment by psarnow** [comment]
I tested this out in my application and its the best way I have found to update polygon 'styles' quickly in leaflet.

**Comment by jrisi256** [comment]
Does anyone know if any more progress has been made on this front? I have a large shapefile (all counties in the 50 United States), and I just want the color to change based on some variable people choose (unemployment, poverty, etc.) rather than having to redraw the entire map every time.
